### ISSUE TITLE
[Bug] 0.5.17: generation degenerates into token loops on Qwen3.5-arch model (works on 0.5.6.dev, with/without DFLASH)

### ISSUE THREAD

**Issue by XiaYiHann** [sgl-project/sglang#35723]
### Bug report

**Model**: orcarouter/Qwen3.8-27B-Uncensored-NVFP4 (arch `Qwen3_5ForConditionalGeneration`, NVFP4 quantized) + z-lab/Qwen3.8-27B-DFlash2 draft, single 80GB GPU.

**Version broken**: 0.5.17 (PyPI wheel). **Version working**: 0.5.6.post3.dev9196+ga4ffb996d (git @a4ffb996d). Same model, same flags, same venv.

**Symptom**: after upgrading to 0.5.17, generation degenerates into single-token repetition loops (e.g. `What What What What …` in content, or `"Let's verify." "Let's verify." …` inside reasoning_content), and never exits thinking / never reaches tool calls. `finish_reason: length`, empty content.

**Isolation already done** (all still reproduce on 0.5.17):
1. With DFLASH (8 draft tokens) → loops
2. Without DFLASH (no spec) → loops
3. Without `--reasoning-parser qwen3` → loops (no reasoning_content now, content loops)
4. Plain chat request, no tools → loops
5. Same requests on 0.5.6.dev9196, identical flags → clean output, tool call returned normally

**Repro** (plain chat, no tools):
```bash
python -m sglang.launch_server \
  --model-path <main-model> --served-model-name qwen3.8-27b \
  --kv-cache-dtype fp8_e4m3 --attention-backend flashinfer \
  --reasoning-parser qwen3 --max-running-requests 8 \
  --mem-fraction-static 0.90 --max-total-tokens 1500000
curl -s localhost:8000/v1/chat/completions -H 'Content-Type: application/json' -d '{
  "model":"qwen3.8-27b","max_tokens":160,"temperature":0.6,
  "messages":[{"role":"user","content":"What is the capital of France? Answer in one short sentence."}]}'
```
Expected: clean answer. Actual (0.5.17): `content: "The user asks for the answer one short sentence.The question asks What What What What What What What What …"`, `finish_reason: length`.

Looks like a generation-level regression in 0.5.17 for this Qwen3.5-based arch (sampling/logits/pipe), independent of speculative decoding. Happy to provide full logs or bisect help.
